src/Microsoft.AspNetCore.Sockets.Http/ConnectionManager.cs

```diff
@@ -94,8 +97,19 @@ public void RemoveConnection(string id)
 
         private static string MakeNewConnectionId()
         {
-            // TODO: We need to sign and encyrpt this
-            return Guid.NewGuid().ToString();
+#if NETCOREAPP2_1
+            // 128 bit buffer / 8 bits per byte = 16 bytes
+            Span<byte> buffer = stackalloc byte[16];
+            _keyGen.GetBytes(buffer);
+            // Generate the id with RNGCrypto because we want a cryptographically random id, which GUID is not
+            return WebEncoders.Base64UrlEncode(buffer.ToArray());
+#else
+            // 128 bit buffer / 8 bits per byte = 16 bytes
+            var buffer = new byte[16];
```

Comment:
Since this buffer is so short-lived and limited in scope, it it worth using `ArrayPool<byte>.Shared` to rent it? Legit not sure exactly what the use cases are for that. @davidfowl ?

Reply:
No I think this is stackalloc worthy but only if we update `WebEncoders.Base64UrlEncode(buffer)` to take a `Span<byte>`